app/build.gradle

```diff
@@ -302,17 +211,12 @@ dependencies {
     }
 
     implementation project(':localeapi')
-    //implementation 'com.nispok:snackbar:2.10.8'
 
-
-    // use specific version of data binding until google resolves the bugs with mixed case package names
```

Comment:
When these databinding versions were changed before, it apparently introduced significant compatibility problems with some handsets and those changes were reverted to resolve that issue. Can this java 11 support be added without altering other dependency versions?

Reply:
> When these databinding versions were changed before, it apparently introduced significant compatibility problems with some handsets and those changes were reverted to resolve that issue. 

The version of the databinding library is defined by the Android Gradle Plugin in the root `build.gradle` and is `classpath 'com.android.tools.build:gradle:3.4.3'`. You can check this if you look at the file `META-INF\com.android.databinding_library.version` in the apk file. The implementations I've removed here are never used. Maybe, they were required for former databinding build tools. Currently, you only need to enable databinding and use the correct AGP.

Hence, we are on 3.4.3. for nearly a year (55c54b367) and on 3.4.1 for nearly two years (54e30f255).

>Can this java 11 support be added without altering other dependency versions?

In general, yes. Just cherry pick 4782eef. But I think it is worth to cleanup the files we have and remove old and redundant code. It is also important to fix at least known bugs in the dependencies by updating to patched/minor versions. Any compatibility issues should be avoided. That's why I check any dependency.

Most of the libraries I've updated are just build time libraries (lombok, gradle). In general, I've just upgraded to a patched version (lombok, google-services). Only New Relic is updated to a newer minor version. I've always linked the change log in the corresponding commit (if any) and checked them carefully.

Reply:
It is unfortunate that databinding is apparently dependent on the gradle version. I hadn't realised that it wasn't respecting the specific implementation versioning and was silently promoting the version. The commit which fixed the databinding related issues that affected certain handsets by reverting the use of V2 and changing the gradle version is acfa86f240006ee595ec82a4a262e5db8f2ff521 from Aug 2021

Reply:
> It is unfortunate that databinding is apparently dependent on the gradle version.

That is true but it is bundled with the Android Gradle Plugin as clearly stated at https://developer.android.google.cn/jetpack/androidx/releases/databinding:

>The databinding library is bundled with [the Android Gradle plugin](https://developer.android.google.cn/studio/releases/gradle-plugin). You do not need to declare a dependency on the library, but you must enable it.
> To enable data binding, set the dataBinding build option to true in your module's build.gradle file, ...

I've confirmed this by checking `META-INF\com.android.databinding_library.version` in the apk file. If you don't trust, verify it by yourself.

I did not touch `android.databinding.enableV2=false` and kept the AGP at 3.4.3.